test/autotest.py

```diff
@@ -509,8 +509,10 @@ def testCheckpoint():
 
     #wait for files to appear and status to return to original
     WAITFOR(lambda: getNumCkptFiles(ckptDir)>0 and \
-                    (CKPT_CMD == 'xc' or doesStatusSatisfy(getStatus(), status)),
+                   (CKPT_CMD == 'xc' or doesStatusSatisfy(getStatus(), status)),
             wfMsg("checkpoint error"))
+    #wait and see if process will die shortly after checkpoint
+    sleep(S*SLOW)
```

Comment:
I hate to do this to you, but this sleep statement doesn't serve the intended purpose. (See my comment below for the updated fix.)

In it's current form, after sleeping for "S*SLOW" seconds, we do not check if the processes are still alive. The `CHECK(doesStatusSatisfy((numFiles,True),status) ...` only checks for a valid number of checkpoint files.

Therefore, we need to insert a `CHECK(doesStatusSatisfy(getStatus(), status))` right after this sleep to ensure that the processes are still alive after ckpt.

Further, as I suggested in one of my previous comments (I can't find it now), this newly added sleep can be removed from here and placed right after the call to testCheckpoint() later in the file.